src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -211,7 +211,7 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
                 // and the property name is required to add the property to the containing type, and
                 // the type and parameters are required to determine the override or implementation.
                 var type = this.ComputeType(bodyBinder, syntax, diagnostics);
-                _lazyType.InterlockedInitialize(type);
+                Interlocked.CompareExchange(ref _lazyType, new TypeWithAnnotations.Boxed(type), null);
```

Comment:
>Interlocked.CompareExchange(ref _lazyType, new TypeWithAnnotations.Boxed(type), null); [](start = 16, length = 86)

It looks like Interlocked operation is not necessary here, we are in a constructor and other threads shouldn't be able to access the instance before it is fully initialized.

Reply:
Andy was suggesting to make the boxed constructor private. So I'm keeping the compare exchange here (and below) despite not being necessary.

---
In reply to: [280895441](https://github.com/dotnet/roslyn/pull/35459#discussion_r280895441) [](ancestors = 280895441)

Reply:
Interlocked is not necessary for initializing fields in a constructor, please remove, it is confusing the reader.

---
In reply to: [280905650](https://github.com/dotnet/roslyn/pull/35459#discussion_r280905650) [](ancestors = 280905650,280895441)